mod/lostpass.php

```diff
@@ -65,74 +60,56 @@ function lostpass_post(App $a) {
 		The login details are as follows:
 
 		Site Location:	%2$s
-		Login Name:	%3$s'));
-
-	$preamble = sprintf($preamble, $username, $sitename);
-	$body = sprintf($body, $resetlink, System::baseUrl(), $email);
+		Login Name:	%3$s', $resetlink, System::baseUrl(), $user['email']));
 
 	notification([
-		'type' => SYSTEM_EMAIL,
-		'to_email' => $email,
-		'subject'=> sprintf( t('Password reset requested at %s'),$sitename),
-		'preamble'=> $preamble,
-		'body' => $body]);
+		'type'     => SYSTEM_EMAIL,
+		'to_email' => $user['email'],
+		'subject'  => t('Password reset requested at %s', $sitename),
+		'preamble' => $preamble,
+		'body'     => $body
+	]);
 
 	goaway(System::baseUrl());
-
 }
 
+function lostpass_content(App $a)
+{
+	$o = '';
+	if (x($_GET, 'verify')) {
+		$pwdreset_token = $_GET['verify'];
```

Comment:
You do now store the token in plaintext in the database, why?

Reply:
What was the reason behind the hashing in the first place?

Reply:
I think it is can be treated like a password - which you shouldn't store unhashed in the database as well.

Reply:
The only problem is if someone get a hold of the database (or the user table or user record) in the issuance window of the token. And at this point it doesn't matter at all if it's hashed or not because the attacker has access to the public and private keys of the user anyway. So yes, the attacker can easily change the password of the user, but it was game over anyway at this point.

The reason we hash the passwords in our database isn't to protect our software, but to protect other software used by the same user with the same password.

Reply:
What we should do though is make password reset tokens invalid after a period of time.

Reply:
I still would like to have it hashed, since it isn't an advantage, having it stored unhashed. We could use the central function that you introduced in this PR.

And yes, we should add some datetime field that stores the date when the token is set invalid.

Reply:
And you still haven't told me why, just a vague hand-waving towards "password-like". I'll add the expiration.

Reply:
I'm working very intuitively - I just have the feeling that it is better, I cannot totally explain. (BTW: My intuition often saved my a...) ;-)

Reply:
This PR, I'm working on it right now.